src/main/java/org/cascadebot/cascadebot/events/ButtonEventListener.java

```diff
@@ -33,15 +36,25 @@ public void onMessageReactionAdd(MessageReactionAddEvent e) {
             if (cache.containsKey(channel.getIdLong())) {
                 if (cache.get(channel.getIdLong()).containsKey(e.getMessageIdLong())) {
                     ButtonGroup group = cache.get(channel.getIdLong()).get(e.getMessageIdLong());
-                    Metrics.INS.buttonsPressed.labels(e.getReaction().getReactionEmote().getName()).inc();
-                    e.getChannel().retrieveMessageById(e.getMessageId()).queue(message -> group.handleButton(e.getMember(), channel, message, e.getReactionEmote()));
-                    e.getReaction().removeReaction(e.getMember().getUser()).queue(); //Idk if we want to allow other reactions on the message
-                    //TODO perms checking
+                    doButtonPress(group, e.getReaction(), e.getMember(), e.getTextChannel());
+                }
+            }
+            if (data.getPersistentButtons().containsKey(channel.getIdLong())) {
+                if(data.getPersistentButtons().get(channel.getIdLong()).containsKey(e.getMessageIdLong())) {
+                    ButtonGroup group = data.getPersistentButtons().get(channel.getIdLong()).get(e.getMessageIdLong());
+                    doButtonPress(group, e.getReaction(), e.getMember(), e.getTextChannel());
                 }
             }
         }
     }
 
+    private void doButtonPress(ButtonGroup group, MessageReaction reaction, Member sender, TextChannel channel) {
+        Metrics.INS.buttonsPressed.labels(reaction.getReactionEmote().getName()).inc();
```

Comment:
Maybe we could add a label about what type of button this is? i.e. persistent or not